### ISSUE TITLE
STFT unit tests fails

### ISSUE THREAD

**Issue by dbogdanov** [jjaner/essentia-musicbricks#5]
Streaming mode:

```
Running tests: ['stft_streaming']
testRegression (synthesis.test_stft_streaming.TestSTFT) ... almostEqualArray: x=-1.36532e-06, y=-1.31529e-06, difference=0.0380361 allowed precision=0.02
FAIL
testWhiteNoise (synthesis.test_stft_streaming.TestSTFT) ... ok
testZero (synthesis.test_stft_streaming.TestSTFT) ... ok

======================================================================
FAIL: testRegression (synthesis.test_stft_streaming.TestSTFT)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/synthesis/test_stft_streaming.py", line 157, in testRegression
    self.assertAlmostEqualVector(outsignal[halfwin:-halfwin], signal[halfwin:-halfwin], self.precision)
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/essentia_test.py", line 111, in assertAlmostEqualVector
    return self.assertTrue(almostEqualArray(found, expected, precision))
AssertionError: False is not true
```

Standard mode:

```
TEST LIST ['stft_standard']
TEST EXCLUDE []
Running tests normally
----------------------------------------------------------------------
Running tests: ['stft_standard']
testRamp (synthesis.test_stft_standard.TestSTFT) ... almostEqualArray: x=0.500006, y=0, difference=0.500006 allowed precision=0.02
FAIL
testRegression (synthesis.test_stft_standard.TestSTFT) ... almostEqualArray: x=0.308118, y=0.301902, difference=0.0205879 allowed precision=0.02
FAIL
testWhiteNoise (synthesis.test_stft_standard.TestSTFT) ... almostEqualArray: x=0.883237, y=0.701311, difference=0.259408 allowed precision=0.02
FAIL
testZero (synthesis.test_stft_standard.TestSTFT) ... ok

======================================================================
FAIL: testRamp (synthesis.test_stft_standard.TestSTFT)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/synthesis/test_stft_standard.py", line 120, in testRamp
    self.assertAlmostEqualVector(outsignal[halfwin:-halfwin], signal[halfwin:-halfwin], self.precision)
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/essentia_test.py", line 111, in assertAlmostEqualVector
    return self.assertTrue(almostEqualArray(found, expected, precision))
AssertionError: False is not true

======================================================================
FAIL: testRegression (synthesis.test_stft_standard.TestSTFT)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/synthesis/test_stft_standard.py", line 137, in testRegression
    self.assertAlmostEqualVector(outsignal[halfwin:-halfwin], signal[halfwin:-halfwin], self.precision)
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/essentia_test.py", line 111, in assertAlmostEqualVector
    return self.assertTrue(almostEqualArray(found, expected, precision))
AssertionError: False is not true

======================================================================
FAIL: testWhiteNoise (synthesis.test_stft_standard.TestSTFT)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/synthesis/test_stft_standard.py", line 105, in testWhiteNoise
    self.assertAlmostEqualVector(outsignal[halfwin:-halfwin], signal[halfwin:-halfwin], self.precision)
  File "/home/dbogdanov/essentia-jjaner/essentia-musicbricks/test/src/unittest/essentia_test.py", line 111, in assertAlmostEqualVector
    return self.assertTrue(almostEqualArray(found, expected, precision))
AssertionError: False is not true

----------------------------------------------------------------------
Ran 4 tests in 0.151s

FAILED (failures=3)
```

**Comment by jjaner** [comment]
The streaming_stft errors are caused by the same problem as here (https://github.com/jjaner/essentia-musicbricks/issues/4).

On the contrary, there is a bug for the standard_ stft test. I already mentioned this issue. It seems a memory problem caused in the overlap add algorithm when called form Python. We can investigate it together.

**Comment by jjaner** [comment]
I synchronized this branch with the latest upstream/master branch in essentia and run the test on Linux (Ubuntu 15.10). The streaming_stft works fine:

```
jjaner@jjaner-VirtualBox:~/XSource/essentia-musicbricks$ python test/src/unittest/all_tests.py stft_streaming
[ WARNING  ] EqualLoudness::signal was not connected to FrameCutter::signal
[ WARNING  ] Cannot disconnect FrameCutter::signal from EqualLoudness::signal as they are not connected
TEST LIST ['stft_streaming']
TEST EXCLUDE []
Running tests normally
----------------------------------------------------------------------
Running tests: ['stft_streaming']
testRegression (synthesis.test_stft_streaming.TestSTFT) ... ok
testWhiteNoise (synthesis.test_stft_streaming.TestSTFT) ... ok
testZero (synthesis.test_stft_streaming.TestSTFT) ... ok
```

**Comment by dbogdanov** [comment]
Now I get another error using this branch: 

```
ValueError: Identifier 'ResampleFFT' not found in registry...
```

when running stft_streaming test. 

ResampleFFT is missing in the code.

**Comment by jjaner** [comment]
The new ResampleFFT algorithm has been added to the repository (in /src/algorithms/synthesis).
Can you try again?

**Comment by dbogdanov** [comment]
Works now, however, the same errors in tests as I reported in the first post. How can I track down an error with OverlapAdd?

**Comment by jjaner** [comment]
I fixed some of the errors in the latest commit (7c4aa4fdb705bf19d793e1bcf6aaef20dd25c27c).
Now the test_stft_standard.py fails only in the second round of the test (compute/reset/compute). Please try this:

> python test/src/unittest/all_tests.py stft_standard

I am not sure if this can be corrected, since the OverlapAdd algorithm actually cannot be reset at every frame since it needs to keep some history frames to reconstruct the signal window correctly.

**Comment by dbogdanov** [comment]
Fixed in aadc0002f23d7ff8097a861dbe0e7b6bd45519e3.
